### PR TITLE
op-migrate: various sanity checks + cleanup

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Adds cleanup to the migration script, ensures that config is sane.

We cannot allow for non monotonic block time. We also try to make eip 1159 stable at start and use a hardcoded initial base fee.

Add additional logging

### CODE REVIEW

File: op-chain-ops/genesis/db_migration.go [hunk 1]
```diff
@@ -54,8 +54,21 @@ func MigrateDB(ldb ethdb.Database, config *DeployConfig, l1Block *types.Block, m
 		}, nil
 	}
 
-	if config.L2GenesisBlockBaseFeePerGas == nil {
-		return nil, errors.New("must configure L2 genesis block base fee per gas")
+	// Ensure EIP 1559 params are sane
+	if config.EIP1559Elasticity == 0 {
```
Comment (mslipper) [inline op-chain-ops/genesis/db_migration.go:58]: Isn't there a `Check()` method for the deploy config? If so, should all these checks go in there?

File: op-chain-ops/genesis/db_migration.go [hunk 2]
```diff
@@ -54,8 +54,21 @@ func MigrateDB(ldb ethdb.Database, config *DeployConfig, l1Block *types.Block, m
...
+		return nil, errors.New("EIP 1559 Elasticity cannot be 0")
+	}
+	if config.EIP1559Denominator == 0 {
+		return nil, errors.New("EIP 1559 Denominator cannot be 0")
+	}
+	if config.L2GenesisBlockGasLimit == 0 {
+		return nil, errors.New("L2 genesis block gas limit cannot be 0")
+	}
+	// Ensure monotonic timestamps
+	if uint64(config.L2OutputOracleStartingTimestamp) <= header.Time {
```
Comment (mslipper) [inline op-chain-ops/genesis/db_migration.go:68]: Nice catch!

**Review by mslipper — APPROVED** [review]
Dropped some non-blocking comments. Were you able to boot the sequencer from a migrated disk after making these changes?

File: op-chain-ops/genesis/db_migration.go (hunk 1 above)
Comment (tynes) [reply op-chain-ops/genesis/db_migration.go:58]: I'll consolidate things there, i was worried it would break other things

**Comment by tynes** [comment]
> Dropped some non-blocking comments. Were you able to boot the sequencer from a migrated disk after making these changes?

Not yet, these are changes from statically analyzing the code, will continue trying the migration and will confirm when its successful

**Comment by mslipper** [comment]
@mergifyio refresh

**Pull request merged**